### PR TITLE
Typo fixes and re-formatting etc.

### PR DESCRIPTION
**Pull request by jfy133**

### CODE REVIEW

File: bare-bones-bash.qmd [hunk 1]
```diff
@@ -308,14 +309,14 @@ Here's how that works, conceptually:
...
-Computer programs can take in and spit out data from different _streams_. By
+Computer programs can take in and spit out data from different _streams_ (@fig-Datastreams). By
 default there are 3 such data streams.
 
 - `stdin` : the **st**an**d**ard **in**put
 - `stdout`: the **st**an**d**ard **out**put
 - `stderr`: the **st**an**d**ard **err**or
 
-![](assets/images/chapters/bare-bones-bash/Datastreams.png){width=80% fig-align="center"}
+![Diagram showing stdin going into the program, and two output streams from the program: stderr and stdout](assets/images/chapters/bare-bones-bash/Datastreams.png){#fig-Datastreams width=80% fig-align="center"}
```
Comment (TCLamnidis) [inline bare-bones-bash.qmd:319]:
```suggestion
![Diagram showing stdin going into the program, and two output streams from the program: stderr and stdout.](assets/images/chapters/bare-bones-bash/Datastreams.png){#fig-Datastreams width=80% fig-align="center"}
```

File: bare-bones-bash.qmd [hunk 2]
```diff
@@ -328,19 +329,19 @@ The `stdout` is the actual output of the command. In some cases this gets
 printed to the screen, but most often this is the information that needs to be
 saved in an output file.  
 The `stderr` is the datastream where errors and warnings go. This gets printed
-to your terminal to let you know when something is not going according to plan!
+to your terminal to let you know when something is not going according to plan (@fig-Datastream_showcase)!
 
-![This program takes no input, and prints one line to the stdout and one line to the stderr.](assets/images/chapters/bare-bones-bash/Datastream_showcase.png){width=100%}
+![This program in the form of a bash script executed in the terminal takes no input, and prints one line to the stdout and one line to the stderr.](assets/images/chapters/bare-bones-bash/Datastream_showcase.png){#fig-Datastream_showcase width=100%}
```
Comment (TCLamnidis) [inline bare-bones-bash.qmd:334]:
```suggestion
![This program (in the form of a bash script executed in the terminal) takes no input, and prints one line to the stdout and one line to the stderr.](assets/images/chapters/bare-bones-bash/Datastream_showcase.png){#fig-Datastream_showcase width=100%}
```

File: bare-bones-bash.qmd [hunk 3]
```diff
@@ -328,19 +329,19 @@ The `stdout` is the actual output of the command. In some cases this gets
...
 
 A "pipe" (`|`) is a really useful feature of `bash` that lets you chain together
 multiple commands! When two commands are chained together with a pipe, the
-`stdout` of the first command becomes the `stdin` of the second!
+`stdout` of the first command becomes the `stdin` of the second (@fig-Piping)!
 The `stderr` is still printed on your screen, so you can always know when things
 fail.
 
-![](assets/images/chapters/bare-bones-bash/Piping.png){width=80% fig-align="center"}
+![Diagram showing stdin going into the program, and two output streams from the program: stderr and stdout, with the stdout being attached to the stdin of a second program](assets/images/chapters/bare-bones-bash/Piping.png){#fig-Piping width=80% fig-align="center"}
```
Comment (TCLamnidis) [inline bare-bones-bash.qmd:344]:
```suggestion
![Diagram showing stdin going into the program, and two output streams from the program: stderr and stdout, with the stdout becoming the stdin of a second program.](assets/images/chapters/bare-bones-bash/Piping.png){#fig-Piping width=80% fig-align="center"}
```

**Review by TCLamnidis — CHANGES_REQUESTED** [review]
BBB edits LGTM, but dark mode is broken. Changes the colour of text but not the background.

**Comment by TCLamnidis** [comment]
The font colour of non-language specific monospace blocks is different between dark and light mode. Since the block has consistent background, I am not sure that is necessary. Maybe both could be a midway between the two colours (#666666 ?) instead.

Also, for some reason swapping between dark/light mode seems to change the new line breaks. This happens both in the sidebar title, but also the text (where it matters less).

**Comment by TCLamnidis** [comment]
Also, I noticed that the portraits of the three authors need updating for darkmode. The white parts of the PNGs are actually transparent, making the dark mode images scarring 😅 . Cannot do that on my end, but someone more skilled with illustrator/Inkscape (@aidaanva) might be able to?

**Comment by aidaanva** [comment]
I can work on it on friday

**Comment by jfy133** [comment]
Feel free to make a new PR if you see anything else cropping up

**Pull request merged**